src/server/transport/pubsub/redis-pub-sub.ts

```diff
@@ -0,0 +1,41 @@
+import type redis from 'redis';
+import { Observable } from 'rxjs';
+import type { GenericPubSub, PubSubChannelId } from './generic-pub-sub';
+import { globalChannelId } from './util';
+
+export class RedisPubSub<T> implements GenericPubSub<T> {
+  private pubClient: redis.RedisClient;
+  private subClient: redis.RedisClient;
+  private subscriptions: Map<string, Observable<T>> = new Map();
+
+  constructor(pubClient: redis.redisclient, subClient: redis.redisclient) {
+    this.pubClient = pubClient;
+    this.subClient = subClient;
+  }
```

Comment:
As discussed on Discord, it seems like it would be helpful to allow this to receive a config object and then create clients automatically. Based off your usage example:

```suggestion
  constructor({
    config,
    pubConfig = config,
    subConfig = config,
    pubClient = redis.createClient(pubConfig),
    subClient = redis.createClient(subConfig),
  }) {
    this.pubClient = pubClient;
    this.subClient = subClient;
  }
```

Reply:
This class is going to be removed from this PR, and moved to its own repo.